appengine_flexible/websockets/main.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample websockets demonstrates an App Engine Flexible app.
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+)
+
+func main() {
+	http.Handle("/", http.FileServer(http.Dir("static")))
+	http.HandleFunc("/ws", socketHandler)
+	http.HandleFunc("/_ah/health", healthCheckHandler)
+	log.Print("Listening on port 8080")
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+// upgrader holds the websocket connection.
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// socketHandler echos websocket messages back to the client.
+func socketHandler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/ws" {
```

Comment:
Is this needed if this handler is only attached to `/ws`? (I don't know what the best practice is with this.)

Reply:
Not sure, deleted.